src/sync/pull.rs

```diff
@@ -221,9 +226,15 @@ pub async fn maybe_end_try_pull(
         .await
         .map_err(GetMainHeadError)?
         .ok_or(MissingMainHead)?;
+
     let main_snapshot = Commit::base_snapshot(&main_head_hash, &dag_read)
         .await
         .map_err(NoBaseSnapshot)?;
+
+    let old_main_head_map = prolly::Map::load(main_snapshot.value_hash(), &dag_read)
```

Comment:
Delay loading this until `pending.is_empty()`? It's not needed in the case we are replaying.

Reply:
Will look into it.